File: LeetCode/2000.Reverse_Prefix_of_Word/Mario/2000.Reverse_Prefix_of_Word.go
func reversePrefix(word string, ch byte) string {
	j := -1
	for i := range word {
		if word[i] == ch {
			j = i
			break
		}
	}

	if j == -1 {
		return word
	}

	result := make([]byte, 0, len(word))
	for i := j; i >= 0; i-- {
		result = append(result, word[i])
	}
	result = append(result, word[j+1:]...)

	return string(result)
}

File: LeetCode/2000.Reverse_Prefix_of_Word/2000.Reverse_Prefix_of_Word.java
class Solution {
    public String reversePrefix(String word, char ch) {
        int j = -1;
        for (int i = 0; i < word.length(); i++) {
            if (word.charAt(i) == ch) {
                j = i;
                break;
            }
        }

        if (j == -1)
            return word;
        char[] chs = word.toCharArray();
        for (int i = 0; i <= j; i++, j--) {
            char tmp = chs[i];
            chs[i] = chs[j];
            chs[j] = tmp;
        }

        return String.valueOf(chs);
    }
}